src/plugins/ui_core_views/cell_evaluation/evaluator.ts

```diff
@@ -275,6 +275,10 @@ export class Evaluator {
       this.invalidateSpreading(position);
     }
 
+    if (this.spreadingRelations.isArrayFormula(position)) {
```

Comment:
alternative idea: move these lines to `evaluator.updateDependencies` and have them out of the evaluation hot path.
It would also be correct, right?